### PR TITLE
Revue du code

### PR DESCRIPTION
**Pull request by mathildebuenerd**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +0,0 @@
-# projet3-reservation-velo
```
Comment (mathildebuenerd) [inline README.md:1]: Tu peux éditer ce fichier en donnant plus de détails sur ton projet
Syntaxe markdown: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

File: index.html [hunk 2]
```diff
@@ -1,133 +0,0 @@
...
-                <a href="#map_container">géolocalisation velo</a>
-            </li>
-            <li class="menu">
-                <a href="#container_canvas">Réservation</a>
-            </li>
-        </ul>
-    </header>
-    <section class="slider_container">
-        <div class="slides" id="slides_1">
-            <!---<img src="icones/icons8-écran-tactile-smartphone-64.png">
```
Comment (mathildebuenerd) [inline index.html:37]: Enlever les lignes commentées avant de rendre le dossier

File: index.html [hunk 3]
```diff
@@ -1,133 +0,0 @@
...
-                <img src="icones/icons8-ordinateur-portable-100.png">
-                <img src="icones/icons8-parking-à-vélos-100.png">
-                <img src="icones/icons8-piste-cyclable-96 (1).png">--->
-            <img src="7797451437_cathedrale-saint-pierre-et-saint-paul-de-nantes.jpg" alt="">
```
Comment (mathildebuenerd) [inline index.html:41]: Textes alternatifs si besoin

File: js/reservation.js [hunk 4]
```diff
@@ -1,138 +0,0 @@
-class Resa {
-    constructor() {
-        this.timeMin = "";
```
Comment (mathildebuenerd) [inline js/reservation.js:3]: Les variables utilisées dans une seule fonction doivent être dans la fonction et pas dans le constructeur

File: js/reservation.js [hunk 5]
```diff
@@ -1,138 +0,0 @@
...
-        if (canvas.validSign === true){
-            //alert("vous de devez signez dans le cadre réservé");
-            console.log("salut");  
-        }
-      
-        else {
-        }
-    }
-    // enregistre les informations de reservation d'un velo
-    InfosStationResa() {
```
Comment (mathildebuenerd) [inline js/reservation.js:37]: jamais de majuscule au nom de fonction

File: js/reservation.js [hunk 6]
```diff
@@ -1,138 +0,0 @@
...
-        this.finResaMin = "";
-        this.finResaHours = "";
-        this.sectionResaElt = document.getElementById("resa");
-        this.infosResaElt = document.getElementById("infos_resa");
-        this.timer = "";
-    }
-    init() {
-        this.valideResa();
-        console.log(canvas.validSign) 
-        let evenement = this.validateButton.addEventListener("click", this.eventButtonValidate.bind(this))
```
Comment (mathildebuenerd) [inline js/reservation.js:21]: si tu ne modifie pas une variable ça peut être `const`

File: js/reservation.js [hunk 7]
```diff
@@ -1,138 +0,0 @@
...
-    eventButtonValidate() {
-        this.time = 1200;
-        this.InfosStationResa();
-        this.heureFinResa();
-        this.recoversInfosResa();
-        this.startTimer();
-        //this.validateButton.setAttribute("disabled", "disabled");
-    }
-    // création de l'heure de fin de resa
-    heureFinResa() {
```
Comment (mathildebuenerd) [inline js/reservation.js:52]: Les noms de fonction devraient commencer par des verbes

File: js/reservation.js [hunk 8]
```diff
@@ -1,138 +0,0 @@
...
-        sessionStorage.setItem("finResaHours", this.finResaHours);
-    };
-    //on affiche les infos de fin de resa enregistrer dans la session 
-    recoversInfosResa() {
-        let dataNom = sessionStorage.getItem("nom");
-        let dataAdresse = sessionStorage.getItem("adresse");
-        let finResaSec = sessionStorage.getItem("finResaSec");
-        let finResaMin = sessionStorage.getItem("finResaMin");
-        let finResaHours = sessionStorage.getItem("finResaHours");
-        this.infosResaElt.textContent = "vous avez réservé un velo à la station " + dataNom + " : veuillez vous rendre a(u)" + dataAdresse + " avant " + finResaHours + " h " + finResaMin + " min " + finResaSec + " s !! " + "il sera réservé pendant les 20 prochaines minutes..";
```
Comment (mathildebuenerd) [inline js/reservation.js:82]: Tu connais les string literal? https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Litt%C3%A9raux_gabarits

```js
const nicolas = "Nicolas";
const salutation = "Hello " + nicolas + " !";
```

peut devenir:

```js
const nicolas = "Nicolas";
const salutation = `Hello ${nicolas} !`;
```

Comment (mathildebuenerd) [reply js/reservation.js:82]: anti quote

File: js/slider.js [hunk 9]
```diff
@@ -1,79 +0,0 @@
...
-        this.arrowLeft = document.getElementById("fa-angle-left");
-        this.arrowRight = document.getElementById("fa-angle-right");
-        this.pause = document.getElementById("pause");
-        this.play = document.getElementById("play");
-        this.index = 0;
-        this.duration = 4000;
-        this.lectureAuto = "";
-    };
-    init() { 
-    this.lectureAuto = setInterval(this.defiler.bind(this), this.duration);
```
Comment (mathildebuenerd) [inline js/slider.js:15]: Indentation automatique

File: js/slider.js [hunk 10]
```diff
@@ -1,79 +0,0 @@
...
-
-    pauseSlides() {
-        clearInterval(this.lectureAuto);
-    }
-    playSlides() {
-        clearInterval(this.lectureAuto);
-        this.lectureAuto = setInterval(this.defiler.bind(this), this.duration);
-    }
-
-    clavier(e) {
```
Comment (mathildebuenerd) [inline js/slider.js:67]: Soit tout en français, soit tout en anglais

File: style.css [hunk 11]
```diff
@@ -1,334 +0,0 @@
...
-
-.slider_container {
-    
-    background: red;
-    width: 700px;
-    height: 480px;
-    /*overflow: hidden;*/
-    margin: auto;
-    margin-top: 30px;
-
```
Comment (mathildebuenerd) [inline style.css:49]: Enlever les espaces en trop

File: index.html [hunk 12]
```diff
@@ -1,133 +0,0 @@
...
-            </div>
-            
-            <div id="places_dispo">
-                    <i class="fas fa-parking"></i>
-                    <p classe="nombre places" id="nombre_places"> : </p>
-            </div>
-            <input id="btn_resa" type="submit" value="Réserver">
-            <section id="canvas_container">
-                    <h3 id="titre_resa">Réservation</h3>
-                    <p>Signez ci-dessus pour valider la réservation d'un velo</p>
```
Comment (mathildebuenerd) [inline index.html:103]: Ajoute des `<label>`  à tes `<input>` pour le nom et le prénom

File: js/ajax.js [hunk 13]
```diff
@@ -1,43 +0,0 @@
...
-            console.error(req.status + " " + req.statusText + " " + url);
-        }
-    });
-    req.addEventListener("error", function () {
-        console.error("Erreur réseau avec l'URL " + url);
-    });
-    req.send(null);
-}
-
-// Exécute un appel AJAX POST
```
Comment (mathildebuenerd) [inline js/ajax.js:20]: Veille à ce que tu comprennes bien ces commentaires

File: js/canvas.js [hunk 14]
```diff
@@ -1,112 +0,0 @@
-class CanvasObjet {
```
Comment (mathildebuenerd) [inline js/canvas.js:1]: Choisir un nom explicite

Ca peut être `Canvas`

File: js/canvas.js [hunk 15]
```diff
@@ -1,112 +0,0 @@
...
-    this.mousePosition = {
-      x: 0,
-      y: 0
-    };
-    this.clearButton = document.getElementById(BtnClear);
-    this.validateButton = document.getElementById(BtnValidate);
-    this.canvas.width = 200;
-    this.canvas.height = 110;
-    this.btnResa = document.getElementById("btn_resa");
-    console.log(this.canvas);
```
Comment (mathildebuenerd) [inline js/canvas.js:19]: Les console.log peuvent être enlevé au moment de rendre

File: js/canvas.js [hunk 16]
```diff
@@ -1,112 +0,0 @@
...
-    console.log(this.ctx);
-  }
-  init() { 
-    this.btnResa.addEventListener("click", this.displayCanvas.bind(this));
-    this.canvas.addEventListener("mousedown", this.mouseDown.bind(this));
-    this.canvas.addEventListener("mousemove", this.mouseMove.bind(this));
-    window.addEventListener("mouseup", this.mouseUp.bind(this));
-    this.clearButton.addEventListener("click", this.clearCanvas.bind(this));
-  }
-  // affichage de la section canvas
```
Comment (mathildebuenerd) [inline js/canvas.js:29]: Pour chaque commentaire, pose-toi la question "Est-ce qu'en simplifiant mon code je pourrais rendre ce commentaire inutile?"

```js
// prénom
const pr = "Nicolas";
```

File: js/canvas.js [hunk 17]
```diff
@@ -1,112 +0,0 @@
...
-       let nomElt = document.getElementById("div_nom");
-       nomElt.style.marginTop = "Opx";
-       let adresseElt = document.getElementById("div_adresse");
-       adresseElt.style.marginTop = "5px";
-       let placesDispoElt = document.getElementById("places_dispo");
-       placesDispoElt.style.marginTop = "10px";
-       let VeloDispoElt = document.getElementById("velos_dispo");
-       VeloDispoElt.style.marginLeft = "50px";
-       VeloDispoElt.style.marginRight = "40px";
-       VeloDispoElt.style.marginTop = "10px";
```
Comment (mathildebuenerd) [inline js/canvas.js:46]: Utiliser plutôt `VeloDispoElt.classList.add('reservation-en-cours');`

```css
.reservation-en-cours {
  margin: 50px 40px 10px;
}
```

Comment (mathildebuenerd) [reply js/canvas.js:46]: classList.remove()

Comment (mathildebuenerd) [reply js/canvas.js:46]: .reservation-en-cours #nom {}
